google-cloud-storage/src/test/java/com/google/cloud/storage/it/ITStorageTest.java

```diff
@@ -3273,4 +3273,45 @@ public void testBlobReload() throws Exception {
     updated.delete();
     assertNull(updated.reload());
   }
+
+  private Blob createBlob(String method, BlobInfo blobInfo) throws IOException {
+    switch (method) {
+      case "create":
+        return storage.create(blobInfo);
+      case "writer":
+        {
+          storage.writer(blobInfo).close();
+          return storage.get(BlobId.of(blobInfo.getBucket(), blobInfo.getName()));
+        }
+      default:
+        throw new IllegalArgumentException("Unknown method " + method);
+    }
+  }
+
+  private void testAutoContentType(String method) throws IOException {
+    String[] names = {"file1.txt", "dir with spaces/Pic.Jpg", "no_extension"};
+    String[] types = {"text/plain", "image/jpeg", "application/octet-stream"};
+    Blob blob = null;
+    for (int i = 0; i < names.length; i++) {
+      BlobId blobId = BlobId.of(BUCKET, names[i]);
+      BlobInfo blobInfo = BlobInfo.newBuilder(blobId).build();
+      blob = createBlob(method, blobInfo);
+      assertEquals(types[i], blob.getContentType());
+    }
+    String customType = "custom/type";
+    BlobId blobId = BlobId.of(BUCKET, names[0]);
+    BlobInfo blobInfo = BlobInfo.newBuilder(blobId).setContentType(customType).build();
+    blob = createBlob(method, blobInfo);
```

Comment:
don't reuse local variable. There are two separate blob objects here that coincidentally share a variable.

Reply:
I had a reason to reuse the variable but after rewriting the test that reason is not actual anymore. Thanks for catching that.